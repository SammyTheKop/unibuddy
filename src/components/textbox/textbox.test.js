import React, { useState } from "react";

// component
import Textbox from "./textbox";

// jest renderer
import renderer from "react-test-renderer";

// testing library react
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

// Case 1: Textbox Renders properly or not
it("Case 1: Textbox Component Renders Correctly or not", () => {
  const buttonComponentTree = renderer
    .create(<Textbox placeholder="This is a Nacho Style Textbox" />)
    .toJSON();
  expect(buttonComponentTree).toMatchSnapshot();
});

/**
 * @name TestComponent
 * @details use case of textbox component
 * @returns button component with the updated count
 */
const TestComponent = () => {
  const [value, setValue] = useState("Test");

  const handleTextboxChange = (inputText) => {
    setValue(inputText);
  };

  return (
    <Textbox
      handleChange={(e) => handleTextboxChange(e.target.value)}
      placeholder="This is a Nacho Style Textbox"
    >
      {value}
    </Textbox>
  );
};

const setup = () => {
  const utils = render(<TestComponent />);
  const input = utils.getByLabelText("textbox");
  return {
    input,
    ...utils,
  };
};

// Case 2: Checks the value of the textbox component
it("Case 2: Checks the value of the textbox component", () => {
  const { input } = setup();
  fireEvent.change(input, {
    target: { value: "Something" },
  });
  expect(input.value).toBe("Something");
});

// Case 3: Checks the value of the textbox component
it("Case 3: Checks the value of the textbox component (Negative Testing)", () => {
  const { input } = setup();
  fireEvent.change(input, {
    target: { value: "Something" },
  });
  expect(input.value).not.toBe("Not Something");
});

// Case 4: Checks the value of the textbox component testid
it("Case 4: Checks the value of the textbox component testid", () => {
  const { getByTestId } = render(<TestComponent />);
  expect(getByTestId("textbox")).toBeInTheDocument();
});

// Case 5: Checks the value of the textbox-clear testid
it("Case 5: Checks the value of the textbox-clear testid", () => {
  const { getByTestId } = render(<TestComponent />);
  expect(getByTestId("textbox")).toBeInTheDocument();
});
