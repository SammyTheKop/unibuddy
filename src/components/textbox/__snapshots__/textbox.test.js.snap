// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Case 1: Textbox Component Renders Correctly or not 1`] = `
<div
  className="textbox-component"
>
  <input
    aria-label="textbox"
    aria-required={false}
    className=""
    data-testid="textbox"
    onChange={[Function]}
    placeholder="This is a Nacho Style Textbox"
    readOnly={false}
    required={false}
    type="text"
    value=""
  />
</div>
`;
